backends/utils_test.py

```diff
@@ -0,0 +1,121 @@
+"""
+Tests for the utils
+"""
+from datetime import datetime, timedelta
+from mock import (
+    MagicMock,
+    patch,
+)
+
+import pytz
+from django.test import TestCase
+from requests.exceptions import HTTPError
+
+from backends import utils
+from profiles.factories import UserFactory
+
+# pylint: disable=protected-access
+
+
+class RefreshTest(TestCase):
+    """Class to test refresh token"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super(RefreshTest, cls).setUpTestData()
+        # create an user
+        cls.user = UserFactory.create()
+        # create a social auth for the user
+        cls.user.social_auth.create(
+            provider='edxorg',
+            uid=cls.user.username,
+            extra_data='{"access_token": "fooooootoken", "refresh_token": "baaaarrefresh"}'
+        )
+
+    def setUp(self):
+        super(RefreshTest, self).setUp()
+        self.now = datetime.now(pytz.utc)
+
+    def update_social_extra_data(self, data):
+        """Helper function to update the python social auth extra data"""
+        social_user = self.user.social_auth.get(provider='edxorg')
+        social_user.extra_data.update(data)
+        social_user.save()
+
+    @patch('backends.edxorg.EdxOrgOAuth2.refresh_token', autospec=True)
+    def test_refresh(self, mock_refresh):
+        """The refresh needs to be called"""
+        extra_data = {
+            "updated_at": (self.now - timedelta(weeks=1)).timestamp(),
+            "expires_in": 100  # seconds
+        }
+        self.update_social_extra_data(extra_data)
```

Comment:
I think there is a misunderstanding here. `update_social_extra_data` does exactly that it says: it updates the `extra_data` field of the `user_social` model instance. I do not create any new instances here; the only place where something is created is in the `setUpTestData` method.

Anyway I think I might simplify the tests by returning the `social_user` object after updating the field.